hollow/src/main/java/com/netflix/hollow/core/index/HollowHashIndex.java

```diff
@@ -160,10 +159,9 @@ private boolean matchIsEqual(FixedLengthElementArray matchHashTable, long hashBu
 
                 HollowObjectTypeReadState objectAccess = (HollowObjectTypeReadState)readState;
                 int fieldIdx = fieldPath[fieldPath.length-1];
-                if(hashOrdinal == -1 && query[i] == null)
-                    continue;
-                if(!HollowReadFieldUtils.fieldValueEquals(objectAccess, hashOrdinal, fieldIdx, query[i]))
+                if(hashOrdinal == -1 || !HollowReadFieldUtils.fieldValueEquals(objectAccess, hashOrdinal, fieldIdx, query[i])) {
```

Comment:
`query[i] == null` is never true due to precheck before:
com/netflix/hollow/core/index/HollowHashIndex.java:89

Another thing, if we leave `continue`, then null value (in our context ordinal with the -1 value) will be interpreted as any. So it will match to any value, like "-1" ~ "one", "-1" ~ "two", ...
For instance, if we'll try to leave `continue`, then `index.findMatches("one:")` from tests will return something, despite the fact, that we didn't put "one:" anywhere.

Reply:
Awesome, got it, thanks!